### PR TITLE
Auth integration

### PR DESCRIPTION
**Pull request by wpears**
First pass at auth integration, depends on https://github.com/cfpb/hmda-platform-auth/pull/19
Know issues:
refreshing a page upsets the flow, ticketed in #258 
uploading a file results in a 400 ticketed in https://github.com/cfpb/hmda-platform-auth/issues/18

### CODE REVIEW

File: src/js/UserManager.js
Comment (hkeeler) [inline src/js/UserManager.js:4]: We discussed setting these values at container startup, likely via some awk/find/replace goodness.  Do you want to do that as part of this PR, or a follow up?  As it is now, this will obviously only work in our local dev setups.

File: src/js/UserManager.js
Comment (hkeeler) [inline src/js/UserManager.js]: You can remove `email`.  Keycloak doesn't honor that scope.

File: src/js/UserManager.js
Comment (hkeeler) [inline src/js/UserManager.js]: Can this be removed now?

File: src/js/api.js
Comment (hkeeler) [inline src/js/api.js]: When do the `/hmda`-based URLs come into play?

File: src/js/containers/Login.jsx
Comment (hkeeler) [inline src/js/containers/Login.jsx]: Same `UserManager` question as above.  This one does seem a bit different, though.  For instance, there's no _silent renew_-related attributes.  Is that intentional?  What's the difference between these two `UserManager`s?

When we do change this, could we create a JSON file that gets read in instead so we don't have to find/replace in-code?  Also, then it'd be setup in a single place.

File: src/js/reducers/index.js
Comment (hkeeler) [inline src/js/reducers/index.js]: Should either remove or comment why it's still here.

**Review by hkeeler — COMMENTED** [review]
All reviewed up.  No show-stoppers.  Mostly questions and thoughts on future improvements.

File: src/js/UserManager.js
Comment (wpears) [reply src/js/UserManager.js:4]: How is the value passed in, then? Environment variable, some discovery endpoint, etc?

File: src/js/api.js
Comment (wpears) [reply src/js/api.js]: Probably never. That was just put in place as a sane-ish default (which might actually be worse because it will be less obvious what's happening in case of failure)

File: src/js/UserManager.js
Comment (hkeeler) [reply src/js/UserManager.js:4]: Yes, envvars.

File: package.json
Comment (awolfe76) [inline package.json]: Minor but I think you can remove the `,`  before the `&&` from `./dist/index.html, &&`.

File: src/js/api.js
Comment (hkeeler) [reply src/js/api.js]: Yeah, seems like it'd be a bit of a head-scratcher if it gets into that state.  Perhaps log an error instead of setting the default?

Comment (awolfe76) [reply src/js/api.js]: Is it from https://github.com/cfpb/hmda-platform-auth/pull/19/files#diff-eb672b1ddd3aa7dae5f2577f38daa271R48?

File: src/js/UserManager.js
Comment (wpears) [reply src/js/UserManager.js:4]: I'll hold this for now, I think it would make a lot of sense to do when we switch to `hmda-platform`'s docker-compose

File: src/js/containers/Login.jsx
Comment (wpears) [reply src/js/containers/Login.jsx]: Yeah this was actually a holdover from a different way of modeling the state that I overlooked. Removed in subsequent commit.

File: src/js/UserManager.js
Comment (hkeeler) [reply src/js/UserManager.js:4]: Cool by me.  Added it to https://github.com/cfpb/hmda-platform/issues/661#issuecomment-257983236.

File: src/js/api.js
Comment (awolfe76) [inline src/js/api.js:6]: Not sure I'm following whats going on with the `noParse` option.

Comment (hkeeler) [reply src/js/api.js:6]: Same.

Comment (wpears) [reply src/js/api.js:6]: The url for the fetch is generally built out of the url of the current page. However, there are some cases where this isn't true, and in those cases the `location` shouldn't be parsed and used for path parts.

Comment (awolfe76) [reply src/js/api.js:6]: So why is that the case for the `/institutions` calls and nothing else?

Comment (wpears) [reply src/js/api.js:6]: So `<server>/1/2017` can be mined for the institution id and the filing period to use in api requests, but a request for individual institutions from the homepage or institutions page can't use anything in the url (this is why the function signature includes the `id` param for `getInstitution`)

File: src/js/api.js
Comment (awolfe76) [inline src/js/api.js:13]: Do we need the `|| 'GET'` since its the default?

File: src/js/api.js
Comment (awolfe76) [inline src/js/api.js:36]: Its not part of this PR but is `obj.submission` ever set?

**Review by hkeeler — APPROVED** [review]
All my auth integration related issues have been resolved or moved to other issues.  @awolfe76, once you're happy with it, merge away.

**Comment by awolfe76** [comment]
__We should hold off merging this!__
Will explain more tomorrow at standup.

File: src/js/api.js
Comment (wpears) [reply src/js/api.js:13]: Aha! I thought the same, but if you pass in your OWN object, the default won't exist! You are setting the _object_ as the default, not the key.

When I first wrote it, I thought the same as you, perhaps this means using a default argument here is worse because of the confusion... that said, who doesn't like default args

File: src/js/api.js
Comment (wpears) [reply src/js/api.js:36]: Not anymore, and you can see that generally the submission function calls include `/submissions/` in their suffixes. I didn't drop it from the code (yet) because I thought we might want submission id in the route at some point. If you'd prefer, I could nix it and just re-add it if we need it later.

Comment (awolfe76) [reply src/js/api.js:36]: Maybe just comment it out with a note.

**Comment by awolfe76** [comment]
Closes #227 and closes #244

File: src/js/api.js
Comment (awolfe76) [reply src/js/api.js:13]: Cool. Its working so 💯 . We can refactor later to make it a bit more clear.

**Review by awolfe76 — APPROVED** [review]
Got it running locally!

**Pull request merged**
